### PR TITLE
Get rid of static state in TxState

### PR DESCRIPTION
**Pull request by andreikoval**

### CODE REVIEW

File: community/kernel/src/main/java/org/neo4j/kernel/impl/api/state/NodeStateImpl.java [hunk 1]
```diff
@@ -41,8 +40,141 @@
...
+        @Override
+        public boolean isPropertyRemoved( int propertyKeyId )
+        {
+            return false;
+        }
+
+        @Override
+        public PrimitiveLongIterator getAddedRelationships( Direction direction )
+        {
+            return null;
```
Comment (MishaDemianenko) [inline community/kernel/src/main/java/org/neo4j/kernel/impl/api/state/NodeStateImpl.java]: next 4 methods should return an empty iterator instead, to make it consistent across methods that return iterators at least

File: community/kernel/src/main/java/org/neo4j/kernel/impl/api/state/TxState.java [hunk 2]
```diff
@@ -353,18 +348,32 @@ public boolean hasChanges()
...
     {
-        return LABEL_STATE.getOrCreate( this, labelId ).getOrCreateNodeDiffSets();
+        if ( labelStatesMap == null )
+        {
+            labelStatesMap = Primitive.intObjectMap();
+        }
+        return labelStatesMap.computeIfAbsent( labelId, unused -> new DiffSets<>() );
+    }
+
+    private ReadableDiffSets<Long> getRoLabelStateNodeDiffSets( int labelId )
```
Comment (MishaDemianenko) [inline community/kernel/src/main/java/org/neo4j/kernel/impl/api/state/TxState.java]: what does that 'Ro' part stand's for here and in other methods?

**Review by MishaDemianenko — CHANGES_REQUESTED** [review]
Couple of small questions/comments

File: community/kernel/src/main/java/org/neo4j/kernel/impl/api/state/TxState.java (hunk 2 above)
Comment (andreikoval) [reply community/kernel/src/main/java/org/neo4j/kernel/impl/api/state/TxState.java]: It meant to be read-only, but I'll remove it.

File: community/kernel/src/main/java/org/neo4j/kernel/impl/api/state/NodeStateImpl.java (hunk 1 above)
Comment (andreikoval) [reply community/kernel/src/main/java/org/neo4j/kernel/impl/api/state/NodeStateImpl.java]: I didn't want to change/break existing contracts, but I agree it should be fixed.

**Review by MishaDemianenko — APPROVED** [review]

**Pull request merged**